src/sql/workbench/contrib/query/browser/fileQueryResultsWriter.ts

```diff
@@ -0,0 +1,336 @@
+import { Disposable } from 'vs/base/common/lifecycle';
+import { isArray } from 'vs/base/common/types';
+import { formatDocumentWithSelectedProvider, FormattingMode } from 'vs/editor/contrib/format/format';
+import { localize } from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ILogService } from 'vs/platform/log/common/log';
+import { Progress } from 'vs/platform/progress/common/progress';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { IUntitledTextEditorService } from 'vs/workbench/services/untitled/common/untitledTextEditorService';
+
+export class FileQueryResultsWriter implements IQueryResultsWriter {
+	private messages: Array<IQueryMessage> = [];
+	private formattedQueryResults: Array<string> = [];
+	private tables: Array<Table<any>> = [];
+	private numQueriesToFormat: number = 0;
+	private queryContainsError: boolean = false;
+	private closingMessageIncluded: boolean = false;
+	private hasCreatedResultsFile: boolean = false;
+	private runner: QueryRunner;
+
+	constructor(
+		runner: QueryRunner,
+		@IInstantiationService private instantiationService: IInstantiationService,
+		@IUntitledTextEditorService private readonly untitledEditorService: IUntitledTextEditorService,
+		@IEditorService private readonly editorService: IEditorService,
+		@ILogService private readonly logService: ILogService,
+		@IConfigurationService private configurationService: IConfigurationService
+	) {
+		this.runner = runner;
+	}
+
+	public onQueryStart() {
+		this.reset();
+	}
+
+	public onResultSet(resultSet: ResultSetSummary | ResultSetSummary[]) {
+		this.numQueriesToFormat++;
+		let resultsToAdd: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultsToAdd = [resultSet];
+		} else {
+			resultsToAdd = resultSet.splice(0);
+		}
+
+		if (this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').results.streaming) {
+			this.addResultSet(resultsToAdd);
+		} else {
+			resultsToAdd = resultsToAdd.filter(e => e.complete);
+			if (resultsToAdd.length > 0) {
+				this.addResultSet(resultsToAdd);
+			}
+		}
+	}
+
+	public async updateResultSet(resultSet: ResultSetSummary | ResultSetSummary[]) {
+		let resultSetSummaries: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultSetSummaries = [resultSet];
+		} else {
+			resultSetSummaries = resultSet.splice(0);
+		}
+
+		if (this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').results.streaming) {
+			for (let resultSetSummary of resultSetSummaries) {
+				let table = this.tables.find(t => t.resultSet.batchId === resultSetSummary.batchId && t.resultSet.id === resultSetSummary.id);
+				if (table) {
+					await table.updateResult(resultSetSummary);
+				} else {
+					this.logService.warn('Got result set update request for non-existant table');
```

Comment:
is there anything else you can add to this message to help determine what went wrong here?

Reply:
No, there isn't unfortunately. I reviewed what we did for the GridPanel when an update request is made to a non-existent table and the same message is logged.